HierarchOrbLB.cpp

```diff
@@ -30,7 +38,11 @@ HierarchOrbLB::HierarchOrbLB(const CkLBOptions &opt): CBase_HierarchOrbLB(opt) {
   // decide which load balancer to call
   // IMPORTANT: currently, the greedy LB must allow objects that
   // are not from existing processors.
+#if CHARM_VERSION > 61002
+  orblb = (CentralLB *)new Orb3dLB_notopo(static_cast<CkMigrateMessage *>(nullptr));
```

Comment:
Minor thing, but you could add a constructor to `Orb3dLB_notopo` that takes no parameters and just calls `init()` like this one does. We only use this `static_cast` of `nullptr` because LBs aren't necessarily guaranteed to have such a constructor, but since you're calling another one of your own LBs, that might make it a bit cleaner.

Reply:
I tried adding "Orb3dLB_notopo() {init();}" and I got the following error:
``
Compiling TreePiece.cpp...
In file included from /astro/users/trq/nchil/src/charm/include/CkIO.decl.h:3:0,
                 from /astro/users/trq/nchil/src/charm/include/ckio.h:8,
                 from ParallelGravity.h:14,
                 from TreePiece.cpp:12:
/astro/users/trq/nchil/src/charm/include/charm++.h: In instantiation of 'CBaseT1<Parent, CProxy_Derived>::CBaseT1(Args&& ...) [with Args = {}; Parent = CentralLB; CProxy_Derived = CProxy_Orb3dLB_notopo]':
Orb3dLB_notopo.h:35:20:   required from here
/astro/users/trq/nchil/src/charm/include/charm++.h:451:63: error: no matching function for call to 'CentralLB::CentralLB()'
   CBaseT1(Args&&... args) : Parent(std::forward<Args>(args)...) { thisProxy = this; }
                                                               ^
/astro/users/trq/nchil/src/charm/include/charm++.h:451:63: note: candidates are:
In file included from MultistepLB.h:20:0,
                 from TreePiece.cpp:16:
/astro/users/trq/nchil/src/charm/include/CentralLB.h:78:3: note: CentralLB::CentralLB(CkMigrateMessage*)
   CentralLB(CkMigrateMessage *m) : CBase_CentralLB(m), concurrent(false) {
   ^
/astro/users/trq/nchil/src/charm/include/CentralLB.h:78:3: note:   candidate expects 1 argument, 0 provided
/astro/users/trq/nchil/src/charm/include/CentralLB.h:73:3: note: CentralLB::CentralLB(const CkLBOptions&)
   CentralLB(const CkLBOptions & opt) : CBase_CentralLB(opt), concurrent(false) { initLB(opt);
   ^
/astro/users/trq/nchil/src/charm/include/CentralLB.h:73:3: note:   candidate expects 1 argument, 0 provided
``

Reply:
Ah, yeah, disregard what I said earlier, we need that argument to match the constructor of the parent class of all LBs, `BaseLB`.